### PR TITLE
googletest: Convert to CMakePackage

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/googletest/package.py [hunk 1]
```diff
@@ -36,15 +36,29 @@ class Googletest(Package):
 
     depends_on("cmake", type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/googletest/package.py]: You can remove this line now. It is already included in `CMakePackage`.

File: var/spack/repos/builtin/packages/googletest/package.py [hunk 2]
```diff
@@ -36,15 +36,29 @@ class Googletest(Package):
...
+            options = ['-DBUILD_GMOCK=OFF', '-DBUILD_GTEST=ON']
+        else:
+            # Old style (contains only GTest)
+            options = []
+        return options
 
-        # Google Test doesn't have a make install
-        # We have to do our own install here.
-        install_tree('include', prefix.include)
+    def install(self, spec, prefix):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/googletest/package.py]: Optionally, you should also be able to do something like this:
```python
@when('@:1.7.0')
def install(self, spec, prefix):
    install_tree(join_path(self.stage.source_path, 'include'),
        prefix.include)

    mkdirp(prefix.lib)
    install('libgtest.a', prefix.lib)
    install('libgtest_main.a', prefix.lib)
```
That way it uses the default `install` method for newer versions and overrides it only for older versions. But it's up to you whether or not you think this is clearer.

**Comment by eschnett** [comment]
Done.

**Comment by adamjstewart** [comment]
You might want to update the PR title too. It looks like 1.8.0 was already in the package.py. You're not adding a new version, just fixing the latest version.

**Pull request merged**